.github/workflows/pre-main.yaml

```diff
@@ -0,0 +1,27 @@
+name: Test Incoming Changes
```

Comment:
We could add a `make` target in this repository that runs the linter with the proposed `.golangci.yaml` config.

Reply:
And, add a new linter test onto [our CI configs](https://github.com/openshift/release/tree/master/ci-operator/config/openshift/cert-manager-operator.), feel free to raise PRs on o/release for this.

That way we do not need to introduce extraneous GitHub Actions (and end up paying for [runners](https://docs.github.com/en/billing/managing-billing-for-your-products/managing-billing-for-github-actions/about-billing-for-github-actions)).

Reply:
This PR uses `ubuntu-latest` runners which are the free tier, no billing plan required.

https://docs.github.com/en/actions/using-github-hosted-runners/using-github-hosted-runners/about-github-hosted-runners#standard-github-hosted-runners-for-public-repositories

As for the make target, I agree we could also add that.  The action `golangci/golangci-lint-action@v7` automatically detects the .golangci.yml and uses those settings and manages the installation and running of the linter for us.